plugins/ata/fu-ata-device.c

```diff
@@ -137,6 +138,50 @@ fu_ata_device_pad_string_for_id (const gchar *name)
 	return g_string_free (str, FALSE);
 }
 
+static gchar *
+fu_ata_device_get_guid_safe (const guint16 *buf, guint16 addr_start)
```

Comment:
Is this the same as the NVMe one? Maybe move into common code?

Reply:
Well this expects a guint16 pointer, so the math to do the checksum will be off if the same is used there.  Probably can come up with a common one though.

Reply:
Thoughts on a location for common code though?  Putting it into `fu-common.c` will need `#ifdef` otherwise efivar becomes a hard dependency.

Reply:
Do we need to use efivars for this? Seems like pretty easy to implement on it's own...

Reply:
moved some of the common stuff into `fu-common-guid`